src/test-run/index.js

```diff
@@ -168,19 +182,80 @@ export default class TestRun extends Session {
 
 
     // Task queue
+    _getNextTaskCommand () {
+        if (this.currentDriverTask) {
+            if (this.currentDriverTask.executeOnServer) {
+                this.currentDriverTask.executeOnServer();
+                return null;
+            }
+
+            return this.currentDriverTask.command;
+        }
+
+        return null;
+    }
+
+    _getExecuteOnServerMethod (command) {
+        return () => {
+            this.inCommandExecutionOnServer = true;
+
+            command
+                .execute()
+                .then(() => {
+                    this._fulfillCurrentDriverTask({});
+                    this.inCommandExecutionOnServer = false;
+
+                    var nextCommand = this._getNextTaskCommand();
+
+                    if (nextCommand && this.pendingRequest) {
+                        nextTick()
+                            .then(() => this._resolvePendingRequest(nextCommand));
+                    }
```

Comment:
Wait command should be in `driverTaskQueue`, but it should be executed on server.
So when server has request for next command and this next command is `wait` we should execute waiting on server (for request server answers `null`). Execution of wait command performed in `executeOnServer` method. After execution of waiting server answer to client wtih next command in `driverTaskQueue`.